Add `LastArrayElement`, `Split`,  and `Trim` types (#159)
sindresorhus/type-fest: readme.md

- [`Entries`](source/entries.d.ts) - Create a type that represents the type of the entries of a collection.
- [`SetReturnType`](source/set-return-type.d.ts) - Create a function type with a return type of your choice and the same parameters as the given function type.
- [`Asyncify`](source/asyncify.d.ts) - Create an async version of the given function type.

> - [`TakeLast`](source/take-last.d.ts) - Extracts the type of the last element of an array.

Reviewer: Since it only works on arrays, I think this name is too generic.
```suggestion
- [`LastArrayElement`](source/take-last.d.ts) - Extracts the type of the last element of an array.
```

?

Author: Applied.

- [`PascalCase`](ts41/pascal-case.d.ts) – Converts a string literal to pascal-case (`FooBar`)
- [`SnakeCase`](ts41/snake-case.d.ts) – Convert a string literal to snake-case (`foo_bar`).
- [`DelimiterCase`](ts41/delimiter-case.d.ts) – Convert a string literal to a custom string delimiter casing.

> - [`QueryResult`](ts41/query-result.d.ts) - Matches element types returned from a query.

Reviewer: "query" of what? Not clear to the user.

Author: Added some clarity.

- [`Split`](ts41/split.d.ts) - Represents an array of strings split using a passed-in character or character set.

> - [`Trim`](ts41/trim.d.ts) - Removes spaces from the edges of a string.

Reviewer: ```suggestion
- [`Trim`](ts41/trim.d.ts) - Remove leading and trailing spaces from a string.
```

### Related

> - [typed-query-selector](https://github.com/g-plane/typed-query-selector) - Enhances `document.querySelector` and `document.querySelectorAll` with a template literal type that matches element types returned from a HTML element query selector.

Reviewer 2: I think *[...] an HTML [...]* is the correct form...

https://english.stackexchange.com/questions/1016/do-you-use-a-or-an-before-acronyms-initialisms
